C/services/common/notification_subscription.cpp

```diff
@@ -94,11 +93,11 @@ void NotificationSubscription::unregisterSubscriptions()
 		// Unregister interest
 		m_storage.unregisterAssetNotification((*it).first,
```

Comment:
You might do better to keep a list of subscriptions to unregister rather than hold the lock when calling another service.